MSn Feature List Builder

Kevin-Murray opened the pull request:
Replaces MS/MS peak list builder and Peak Extender modules.
 - MS/MS peak list builder needed an update to include MSn scans. I added a simple feature builder to remove the need for the peak extender module. 

Modules builds simple modular features for precursor ions with MSn (MS2 or greater) fragmentation scans. 

Added feature util to build simple modular feature for in mz and rt range. Please let me know if there is a better way to do this.

tomas-pluskal commented:
Thanks, Kevin!
If the peak extender module is not needed, please remove it as part of this pull request.
Also, I would like to ask you to write more details into the help file:
- Describe the meaning of the parameters more clearly. What is "MS level for precursor chromatogram building"? How does it work? Why the user should not set the MS level in the Scan selection? What happens if he does?
- Describe the algorithm how the chromatogram of the precursor peak is constructed

Kevin-Murray commented:
Thank you for the feedback!

I attempted to improve the clarity of the module help file. Please let me know if it is still lacking. Fundamentally, the module builds an extracted ion chromatogram for each MSn-containing precursor in the specified ranges. 

The MS level parameter is still a bit awkward. I have to set this parameter independent of the scan selection parameter because of how the fragment scan precursor information is stored. Only the precursor MZ variable is stored, not the precursor scan number. So I have to trace MSn scans from the bottom up. For example, to find the precursor of an MS4 scan - I find MS2 scans that have MS3 fragment scans and then search those MS3 scans for MS4 fragment scans. It works, but it's not the most intuitive from a user perspective.

tomas-pluskal commented:
Hi Kevin, this explanation about the meaning of the MS level parameter is good, but I think the explanation should be included in the help file. The more details you provide about the implementation, the better.

Kevin-Murray commented:
I updated the description to reflect the algorithm and why the MS Level parameter must not be set in the scan selection field. 

I also removed a method using the ExtendedPeak class I forgot in the last round of commits.

tomas-pluskal commented:
Thanks Kevin, I think it looks good now!
